website/blog/2025-09-29-blog-post-title.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Bringing NumPy's type-completeness score to nearly 90%
+description: We tell the story of how we brought NumPy's type-completeness score from ~33% to nearly 90%
+slug: numpy-type-completeness
+authors: [marcogorelli]
+tags: [typechecking, news]
+hide_table_of_contents: false
+---
+
+[NumPy](https://numpy.org/) is one of the most downloaded packages in the Python ecosystem. Therefore, any incremental improvement can have a large impact on the data science ecosystem. In particular, improvements related to static typing can improve developer experience and help downstream libraries write safer code. We'll tell the story about how we helped bring its type-completeness score to nearly 90% from an initial 33%.
+
+**TL;DR**:
+
+- NumPy's type-completeness score was ~33%.
+- A one-line fix doubled coverage to over 80%.
+- Fully typing MaskedArray pushed the score to nearly 90%.
+- What's left? Top-level numpy.ma functions, more precise overloads, and adding a type-checker to NumPy's CI.
+
+## Wait, what's type completeness?
+
+Modern IDEs use type annotations to help developers by showing them helpful suggestions about which methods may be available on their variables, as well as to facilitate syntax highlighting. [Pyright](https://github.com/microsoft/pyright) is a popular type-checker which, as well as checking for correctness and consistency, can also measure what percentage of a library's public API has type annotations. We call the percentage of fully-typed symbols exported by a library the _type-completeness score_.
```

Comment:
"Modern IDEs use type annotations to help developers by showing them helpful suggestions and highlighting syntax."

(If you were going to go into detail, you'd need to list more features than just "method discovery", but I think the shorter statement is better.)